packages/client/src/runtime/core/engines/library/LibraryEngine.ts

```diff
@@ -348,7 +337,11 @@ You may have to run ${green('prisma generate')} for your changes to take effect.
 
   on(event: EngineEventType, listener: (args?: any) => any): void {
     if (event === 'beforeExit') {
-      this.beforeExitListener = listener
+      console.warn(
+        `${yellow(
+          'prisma:warn',
+        )} "beforeExit" hook is not applicable to the library engine type, it is only relevant and implemented for the binary engine type. Please add your event listener to the \`process\` object directly instead.`,
```

Comment:
```suggestion
        )} "beforeExit" hook is not applicable to the library engine type since Prisma 5.0.0, it is only relevant and implemented for the binary engine type. Please add your event listener to the \`process\` object directly instead.`,
```

Reply:
Should we have a hard error here? `beforeExit` event will never work, `console.warn` might be pretty easy to miss.